drivers/adc/ad9656/ad9656.c

```diff
@@ -0,0 +1,313 @@
+
+	if (ret != SUCCESS)
+		return ret;
+
+	if (test_mode == AD9656_TEST_OFF)
+		format = AD9656_FORMAT_OFFSET_BINARY;
+	else
+		format = AD9656_FORMAT_2S_COMPLEMENT;
+
+	return ad9656_reg_write(dev, AD9656_REG_OUTPUT_MODE, format);
+}
+
+/**
+ * @brief Launch the user input test functionality for the receive path for the ad9656 chip.
+ * 	      User input data is supplied on the outputs instead of real data captured on the ADC.
+ * @param dev - The device handler for the ad9656 chip
+ * @param test_mode - The type of test that is to be performed or OFF if the testing
+ * 					  process is to be stopped
+ * @param user_input_test_pattern - User input test pattern that is sent on the output instead of
+ * 					the ADC	data
+ * @return SUCCESS if the ad9656 chip could be successfully set for user input testing, FAILURE
+ * 		   otherwise
+ */
+int32_t ad9656_user_input_test(struct ad9656_dev *dev, uint32_t test_mode,
+			       struct ad9656_user_input_test_pattern user_input_test_pattern)
+{
+	int32_t ret;
+
+	/* write user input test pattern in the registers */
+	ret = ad9656_reg_write(dev, AD9656_REG_USER_TEST_PATTERN_1_LSB,
+			       user_input_test_pattern.user_test_pattern1 & 0xFF);
+
+	if (ret != SUCCESS)
+		return ret;
+
+	ret = ad9656_reg_write(dev, AD9656_REG_USER_TEST_PATTERN_1_MSB,
+			       user_input_test_pattern.user_test_pattern1 >> 8);
+
+	if (ret != SUCCESS)
+		return ret;
+
+	ret = ad9656_reg_write(dev, AD9656_REG_USER_TEST_PATTERN_2_LSB,
+			       user_input_test_pattern.user_test_pattern2 & 0xFF);
+
+	if (ret != SUCCESS)
+		return ret;
+
+	ret = ad9656_reg_write(dev, AD9656_REG_USER_TEST_PATTERN_2_MSB,
+			       user_input_test_pattern.user_test_pattern2 >> 8);
+
+	if (ret != SUCCESS)
+		return ret;
+
+	/* determine the chip to output the user test pattern */
+	return ad9656_reg_write(dev, AD9656_REG_ADC_TEST_MODE, test_mode);
+}
+
+/**
+ * @brief Setup the working parameters of the ad9656 chip
+ * @param device - The device handler of the ad9656 chip
+ * @param init_param - Values for the working parameters of ad9656
+ * @return SUCCESS if device is ready for use, FAILURE otherwise
+ */
+int32_t ad9656_setup(struct ad9656_dev **device,
+		     const struct ad9656_init_param *init_param)
+{
+	uint8_t chip_id;
+	uint8_t pll_stat;
+	int32_t ret;
+	struct ad9656_dev *dev;
+	uint8_t tmp;
+
+	dev = (struct ad9656_dev *)calloc(1, sizeof(*dev));
+	if (!dev)
+		return FAILURE;
+
+	/* SPI */
+	ret = spi_init(&dev->spi_desc, &init_param->spi_init);
+	if (ret != SUCCESS)
+		return ret;
+
+	ret = ad9656_reg_write(dev, AD9656_SPI_CONFIG, 0x3C);	/* RESET */
+
+	if (ret != SUCCESS)
+		return ret;
+
+	mdelay(250);
+
+	ret = ad9656_reg_read(dev, AD9656_REG_CHIP_ID, &chip_id);
+
+	if (ret != SUCCESS)
+		return ret;
+
+	if (chip_id != AD9656_CHIP_ID) {
+		printf("AD9656: Invalid CHIP ID (0x%x).\n", chip_id);
+		return FAILURE;
+	}
+
+	ret = ad9656_reg_write(dev, AD9656_REG_LINK_CONTROL,
+			       0x15);	/* disable link, ilas enable */
+
```

Comment:
no need for extra line here. These two go together without an empty line separating them.

Reply:
Deleted all the extralines in v8